File: python/get_article_html.py
#-- title: Get Article HTML
#-- description: Code to get Medium Article in plain HTML format.
#-- tags: python, medium_api, medium_api_py


# Import libraries
import os
from medium_api import Medium

#%%
# Get RAPIDAPI_KEY from the environment
api_key = os.getenv('RAPIDAPI_KEY')

#%%
# Create a `Medium` Object
medium = Medium(api_key)

#%%
# Get the "Article" object
article = medium.article(article_id="9822a049e763")

#%%
# Save the Article's HTML, as full webpage, with css stylesheet, and print it
article.save_html(
    fullpage=True, 
    style_file="https://mediumapi.com/styles/dark.css"
)

print(article.html)

File: javascript/getArticleHTML.js
// getArticleHTML.js - Fetches the HTML content of a specific article

import Medium from 'medium-api-js';
import dotenv from 'dotenv';

dotenv.config();

const medium = new Medium(process.env.RAPIDAPI_KEY);

// Specify the article ID for which HTML content needs to be fetched
const articleId = '67fa62fc1971'; 

// Set the parameters for full_page and style_file
const fullPage = true;  // Set to true if you want the full page
const styleFile = "dark.css"; // Set the css file name if you want to include the style file

// Fetch the HTML content of the article and log the response
medium.getArticleHTML(articleId, fullPage, styleFile)
    .then(data => {
        console.log('Article HTML:', data);
    })